Error building automake

On a Fedora 25 system, `spack install automake` fails with an error during the build phase. The build log ends with:

```
<snip>
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating t/wrap/aclocal-1.15
config.status: creating t/wrap/automake-1.15
config.status: linking <$HOME>/spack/var/spack/stage/automake-1.15-bhojsqx6uunp35vog7p2v7nmgcvypb36/automake-1.15/GNUmakefile to GNUmakefile
==> 'make' '-j8'
  GEN      bin/automake
  GEN      bin/aclocal
  GEN      t/ax/cc-no-c-o
  GEN      doc/aclocal.1
  GEN      runtest
  GEN      doc/automake.1
  GEN      t/ax/shell-no-trail-bslash
  GEN      lib/Automake/Config.pm
  GEN      t/ax/test-defs.sh
  GEN      doc/aclocal-1.15.1
  GEN      doc/automake-1.15.1
help2man: can't get `--help' info from automake-1.15
Try `--no-discard-stderr' if option outputs to stderr
Makefile:3687: recipe for target 'doc/automake-1.15.1' failed
make: *** [doc/automake-1.15.1] Error 2
make: *** Waiting for unfinished jobs....

```

On a recent `develop` on CentOS 7, I can build `spack install automake` w/out any problems.

If I add @tgamblin's `depends_on('perl', type=('build', 'run'))` then I get this failure:

```
[...]
  GEN      doc/aclocal-1.15.1
help2man: can't get `--help' info from aclocal-1.15
Try `--no-discard-stderr' if option outputs to stderr
make: *** [doc/aclocal-1.15.1] Error 126
make: *** Waiting for unfinished jobs....
```

@adamjstewart -- have I reproduced the problem you saw or discovered something new?

--- a/var/spack/repos/builtin/packages/autoconf/package.py
+++ b/var/spack/repos/builtin/packages/autoconf/package.py
@@ -39,8 +39,29 @@ class Autoconf(AutotoolsPackage):
     # Note: m4 is not a pure build-time dependency of autoconf. m4 is
     # needed when autoconf runs, not only when autoconf is built.
     depends_on('m4@1.4.6:', type=('build', 'run'))
+    depends_on('perl', type=('build', 'run'))
 
     build_directory = 'spack-build'
+
+    def patch(self):
+        # The full perl shebang might be too long; we have to fix this here
+        # because autom4te is called during the build
+        filter_file('^#! @PERL@ -w',
+                    '#! /usr/bin/env perl',
+                    'bin/autom4te.in')
+
+    @run_after('install')
+    def filter_sbang(self):
+        # We have to do this after install because otherwise the install
+        # target will try to rebuild the binaries (filter_file updates the
+        # timestamps)
+        perl = join_path(self.spec['perl'].prefix.bin, 'perl')
+
+        # Revert sbang, so Spack's sbang hook can fix it up
+        filter_file('^#! /usr/bin/env perl',
+                    '#! {0} -w'.format(perl),
+                    '{0}/autom4te'.format(self.prefix.bin),
+                    backup=False)
 
     def _make_executable(self, name):
         return Executable(join_path(self.prefix.bin, name))

--- a/var/spack/repos/builtin/packages/automake/package.py
+++ b/var/spack/repos/builtin/packages/automake/package.py
@@ -36,8 +36,16 @@ class Automake(AutotoolsPackage):
     version('1.11.6', '0286dc30295b62985ca51919202ecfcc')
 
     depends_on('autoconf', type='build')
+    depends_on('perl', type=('build', 'run'))
 
     build_directory = 'spack-build'
+
+    def patch(self):
+        # The full perl shebang might be too long
+        for file in ('aclocal', 'automake'):
+            filter_file('^#!@PERL@ -w',
+                        '#!/usr/bin/env perl',
+                        't/wrap/{0}.in'.format(file))
 
     def _make_executable(self, name):
         return Executable(join_path(self.prefix.bin, name))

--- a/var/spack/repos/builtin/packages/automake/package.py
+++ b/var/spack/repos/builtin/packages/automake/package.py
@@ -31,6 +31,7 @@ class Automake(AutotoolsPackage):
     homepage = 'http://www.gnu.org/software/automake/'
     url      = 'http://ftp.gnu.org/gnu/automake/automake-1.15.tar.gz'
 
+    version('1.15.1', '95df3f2d6eb8f81e70b8cb63a93c8853')
     version('1.15',   '716946a105ca228ab545fc37a70df3a3')
     version('1.14.1', 'd052a3e884631b9c7892f2efce542d75')
     version('1.11.6', '0286dc30295b62985ca51919202ecfcc')

